src/twisted/words/test/test_jabberjstrports.py

```diff
@@ -9,7 +9,8 @@
 from twisted.trial import unittest
 
 from twisted.words.protocols.jabber import jstrports
-from twisted.application.internet import TCPClient
+from twisted.application.internet import TCPClient  # type: ignore[attr-defined]  # noqa
```

Comment:
this is ok... the code from t.application.internet is ingenious  and we can keep it... I would vote for refactoring it to make it easier to read... for multicast and UDP we only have server-side applicitons.

below is the code from application/internet.py
```
_doc={
'Client':
"""Connect to %(tran)s

Call reactor.connect%(tran)s when the service starts, with the
arguments given to the constructor.
""",
'Server':
"""Serve %(tran)s clients

Call reactor.listen%(tran)s when the service starts, with the
arguments given to the constructor. When the service stops,
stop listening. See twisted.internet.interfaces for documentation
on arguments to the reactor method.
""",
}

for tran in 'TCP UNIX SSL UDP UNIXDatagram Multicast'.split():
    for side in 'Server Client'.split():
        if tran == "Multicast" and side == "Client":
            continue
        if tran == "UDP" and side == "Client":
            continue
        base = globals()['_Abstract'+side]
        doc = _doc[side] % vars()

        klass = type(tran+side, (base,), {'method': tran, '__doc__': doc})
        globals()[tran+side] = klass
```

Reply:
OK, I removed the change for this PR, and refactored the code here: https://github.com/twisted/twisted/pull/1339